### PR TITLE
Fix: avoid mutating nextConfig

### PR DESCRIPTION
**Pull request by davesnx**
While updating to beta.10, I got a TypeError on i18n.

This PR 'solves' the issue, but gives back a bunch of json-schema validation errors. I preferred to create a PR half-backed than open an issue.

```
TypeError: Cannot create property 'i18n' on string 'phase-development-server'
    at withNextra (/Users/davesnx/Code/github/davesnx/styled-ppx/packages/website/node_modules/nextra/dist/index.js:207:21)
    at Object.normalizeConfig (/Users/davesnx/Code/github/davesnx/styled-ppx/packages/website/node_modules/next/dist/server/config-shared.js:123:18)
    at Object.loadConfig [as default] (/Users/davesnx/Code/github/davesnx/styled-ppx/packages/website/node_modules/next/dist/server/config.js:74:53)
    at async NextServer.prepare (/Users/davesnx/Code/github/davesnx/styled-ppx/packages/website/node_modules/next/dist/server/next.js:115:24)
    at async /Users/davesnx/Code/github/davesnx/styled-ppx/packages/website/node_modules/next/dist/cli/next-dev.js:147:9
error Command failed with exit code 1.
```

Moved the file to TypeScript to avoid future problems. Not sure if it's desired, but I found myself doing something silly without TS, even in this PR.

### CODE REVIEW

**Comment by davesnx** [comment]
I'm unable to see the logs for **Vercel – nextra-theme-docs-dev**

**Review by promer94 — CHANGES_REQUESTED** [review]
Could you keep the js extensions and use commonjs export. 
Otherwise it would cause some compatibility issues

**Comment by davesnx** [comment]
Done, but I would love to know the reason why it can't be defined under tsup.config.js as `format: 'commonjs',`?

File: packages/nextra/src/index.js [hunk 1]
```diff
@@ -4,37 +4,41 @@ import { DEFAULT_LOCALE, MARKDOWN_EXTENSION_REGEX } from './constants'
...
+      locales: [DEFAULT_LOCALE],
+      defaultLocale: DEFAULT_LOCALE,
+    }
+    // If `i18n.locales` and `i18n.defaultLocale` were not specified,
+    // client will receive error - Text content does not match server-rendered HTML.
+    // Due to `const { locale } = useRouter()` where `locale` will be `undefined`
+    // To fix it we need to explicitly specify `i18n.locales` and `i18n.defaultLocale`
+    const i18n = {
+      locales: nextConfig.i18n?.locales || defaultI18n.locales,
+      defaultLocale: nextConfig.i18n?.defaultLocale || defaultI18n.defaultLocale
```
Comment (B2o5T) [inline packages/nextra/src/index.js:36]:
```suggestion
    // If `i18n.locales` and `i18n.defaultLocale` were not specified,
    // client will receive error - Text content does not match server-rendered HTML.
    // Due to `const { locale } = useRouter()` where `locale` will be `undefined`
    // To fix it we need to explicitly specify `i18n.locales` and `i18n.defaultLocale`
    const i18n = {
      locales: nextConfig.i18n?.locales || [DEFAULT_LOCALE],
      defaultLocale: nextConfig.i18n?.defaultLocale || DEFAULT_LOCALE
```

**Review by promer94 — APPROVED** [review]

**Pull request merged**